### ISSUE TITLE
Implement unary operator execution

### ISSUE THREAD

**Issue by Razican** [jasonwilliams/boa#321]
I just noticed that we haven't implemented all unary operators in execution.

 - [x] `UnaryOp::IncrementPost` (`a++`)
 - [x] `UnaryOp::IncrementPre` (`++a`)
 - [x] `UnaryOp::DecrementPost` (`a--`)
 - [x] `UnaryOp::DecrementPre` (`--a`)
 - [x] `UnaryOp::Minus` (`-a`)
 - [x] `UnaryOp::Plus` (`+a`)
 - [x] `UnaryOp::Not` (`!a`)
 - [x] `UnaryOp::Tilde` (`~a`)
 - [ ] `UnaryOp::Delete` (`delete`)
 - [ ] `UnaryOp::Void` (`void`)

This is implemented in `boa/src/exec/mod.rs`, around line [296][lines].

I think that the "pre" increment/decrement operators, along with the `typeof` and `void` operators should be easy to implement, and good for newcomers to Boa.

The "post" increment/decrement operators will be challenging, as we currently always fully evaluate an AST node before continuing, and these need to be evaluated afterwards. The `delete` operator will be challenging too, I think.

After this is done, in #304 I have implemented some tests that are currently ignored, and we can probably start running them.

[lines]: https://github.com/jasonwilliams/boa/blob/master/boa/src/exec/mod.rs#L296-L314

**Comment by jasonwilliams** [comment]
Did you want to close https://github.com/jasonwilliams/boa/issues/118 in favour of this?

**Comment by Razican** [comment]
> Did you want to close #118 in favour of this?

Actually, I didn't notice that one, but I think there are two things in there. That issue showed that the parsing was broken. That has been solved. Now the issue is not a bug, it's that we don't have that implemented.

So I would say we should close the issue, as the bug is already fixed, and that we should implement the execution.

I already did a small change in #304, and changed the `unreachable!()` for an `unimplemented!()`. It should give a better error message.
